cmd/restic/progress.go

```diff
@@ -29,8 +29,8 @@ func calculateProgressInterval(show bool, json bool) time.Duration {
 	return interval
 }
 
-// newProgressMax returns a progress.Counter that prints to stdout.
-func newProgressMax(show bool, max uint64, description string) *progress.Counter {
+// newTerminalProgressMax returns a progress.Counter that prints to stdout or terminal if provided.
+func newTerminalProgressMax(show bool, max uint64, description string, term *termstatus.Terminal) *progress.Counter {
```

Comment:
I think a different method signature would be much cleaner:

`func newGenericProgressMax(show bool, max uint64, description string, print func(status string)) *progress.Counter` (the call to `printProgress` is then replaced with `print(status)`)

Then `newTerminalProgressMax` would look like the following:
```go
func newTerminalProgressMax(show bool, max uint64, description string, term *termstatus.Terminal) *progress.Counter {
    return newGenericProgressMax(show, max, description, func(status string) {
        term.SetStatus([]string{status})
    })
}
```

A similar conversion also works for `newProgressMax`